engine/storage/src/org/apache/cloudstack/storage/image/BaseImageStoreDriverImpl.java

```diff
@@ -79,6 +80,25 @@
     VMTemplateZoneDao _vmTemplateZoneDao;
     @Inject
     AlertManager _alertMgr;
+    @Inject
+    ImageStoreDetailsDao _imageStoreDetailsDao;
+
+    private static final String NFS_VERSION_DETAILS_KEY = "nfs.version";
```

Comment:
I know this was added as part of a previous commit, but not sure if this the right place to put nfs version related details. What if the image store type is not NFS but SWIFT/S3/CIFS? I think there needs to be a better abstraction of the properties based on store type.